pkg/solana/txm/pendingtx.go

```diff
@@ -536,6 +591,9 @@ func (c *pendingTxContext) OnError(sig solana.Signature, retentionTimeout time.D
 func (c *pendingTxContext) GetTxState(id string) (utils.TxState, error) {
```

Comment:
The error return type here is a misnomer.
It just means that Tx not found. The calling code too just ingores the error field if error != nil.

Can we use same pattern that a map uses, when you try to access a key that doesn't exist?
Instead of error, have a boolean ok, which will be set to false when not found.

if tx, exists := GetTxState(id); exists {
  // tx already exists
  return tx
}

Reply:
like this